chain/sync.go

```diff
@@ -1476,8 +1486,16 @@ func (syncer *Syncer) iterFullTipsets(ctx context.Context, headers []*types.TipS
 			}
 		}
 		i -= batchSize
+
+		if i >= 0 {
+			windowSize += 10
+			log.Infof("successfully fetched %d messages; increasing window size to %d", len(bstout), windowSize)
+		}
```

Comment:
Should make sure to not go over `blocksync.MaxRequestLength` (finality / 900 blocks)

Reply:
ah, good point -- I wasn't aware of that limit! fixing.

Reply:
fixed